mysite/polls/admin.py

```diff
@@ -1,14 +1,61 @@
+from django import forms
 from django.contrib import admin
+# from django.contrib import messages
 
 from .models import Choice, Question
 
+import yaml
+
+
+def contains_curse_words(entry):
+    with open("polls/curse_words.yaml", 'r') as stream:
+        curse_words = yaml.safe_load(stream)
+    my_list = []
+    for curse_word in curse_words:
+        if curse_word in entry.lower():
+            my_list.append(curse_word)
+    return my_list
+
+# def validate_no_curse_words(field, entry, form):
+#     flag = True
+#     with open("polls/curse_words.yaml", 'r') as stream:
+#         curse_words = yaml.safe_load(stream)
+#     for curse_word in curse_words:
+#         if curse_word in entry.lower():
+#             msg = "Please don't use curse word: " + curse_word
+#             form.add_error(field, msg)
+#             flag = False
+#     return flag
+
+
+class QuestionAdminForm(forms.ModelForm):
+    def clean(self):
+        cleaned_data = super().clean()
+        field = "question_text"
+        entry = cleaned_data.get(field)
+        for curse_word in contains_curse_words(entry):
+            msg = "Please don't use curse word: " + curse_word
+            self.add_error(field, msg)
+
+
+class ChoiceAdminForm(forms.ModelForm):
+    def clean(self):
+        cleaned_data = super().clean()
+        field = "choice_text"
+        entry = cleaned_data.get(field)
```

Comment:
More of a suggestion, but is there a way to remove the repetition between the clean method here and in the `QuestionAdminForm`?

Reply:
When I abstracted this functionality into a single function (the contains_curse_words function), I was running into problems trying to unit test that function. Do you think I should attempt the way I was doing it before?

Reply:
I see; I think @aivu had stopped by and discussed this with you already and I reviewed your Slack message from earlier that involves this particular rewrite. I think this is fine.

Reply:
She did, but I am a little bugged by this. If anyone has any elegant solutions please let me know. Otherwise for now I think it's best to focus on the other issues.